velox/common/base/BloomFilter.h

```diff
@@ -31,9 +34,27 @@ namespace facebook::velox {
 // expected entry, we get ~2% false positives. 'hashInput' determines
 // if the value added or checked needs to be hashed. If this is false,
 // we assume that the input is already a 64 bit hash number.
-template <bool hashInput = true>
+// case:
+// InputType can be one of folly hasher support type when hashInput = false.
+// InputType can only be uint64_t when hashInput = true.
+template <class InputType = uint64_t, bool hashInput = true>
 class BloomFilter {
  public:
+  explicit BloomFilter(HashStringAllocator* allocator)
```

Comment:
Template the allocator and default to `std::allocator<uint64_t>`.  See https://github.com/facebookincubator/velox/blob/8f87cefc26dc3cbfb7d081630840aedf8d6e59f7/velox/functions/lib/KllSketch.h#L47-L55

Reply:
The allocator is initialized in agg function, like https://github.com/facebookincubator/velox/blob/main/velox/common/hyperloglog/SparseHll.h#L26
We cannot init StlAllocator by template.

Reply:
You can, check https://github.com/facebookincubator/velox/blob/8f87cefc26dc3cbfb7d081630840aedf8d6e59f7/velox/functions/prestosql/aggregates/ApproxPercentileAggregate.cpp#L38-L44